src/domain/email/entities/create-email-message.dto.entity.ts

```diff
@@ -1,7 +1,7 @@
 export class CreateEmailMessageDto {
-  to: string[];
-  template: string;
-  subject: string;
-  substitutions: Record<string, unknown> | null;
+  to!: string[];
```

Comment:
I don't have a strong opinion regarding this, I've never faced this dilemma before, but adding the non-null assertion operator (`!`) to each property in each class seems a bit like a workaround 🤔 

I guess a more _permanent_ solution could add constructors to the classes and specify the access modifier on each property. That would also force us to review which classes can be converted to interfaces or types. 

Of course this wouldn't be covered in this PR, just food for thought. What do you think @iamacook @fmrsabino ?

Reply:
Nevermind, I've just seen @fmrsabino comment here: https://github.com/safe-global/safe-client-gateway/pull/1051/files#r1463236193, I think it reflects the same concern 🙂

Reply:
I am impartial. I think we can add constructors later if we prefer?

For now, I think the assertion is the "quickest" solution and is [the recommendation of the NestJS author](https://github.com/nestjs/nest/issues/4178#issuecomment-614006219).

Reply:
Sure, I agree the solution is sufficient for now. Later we can revisit this and think about the constructors or even migrating some classes to types/interfaces if it's feasible. All good from my side!

Reply:
> I am impartial. I think we can add constructors later if we prefer?
> 
> For now, I think the assertion is the "quickest" solution and is [the recommendation of the NestJS author](https://github.com/nestjs/nest/issues/4178#issuecomment-614006219).

Yes. I'd add these in a separate PR.